### PR TITLE
Auth Update for nginx.md

### PR DESCRIPTION
**Pull request by cmachikas**
Updating documentation with a walkthrough on simple httpd auth using nginx. 

## Description
Added an additional section under nginx walkthroughs. 

## Motivation and Context
Solves a major feature request of many end-users through existing tools. 

## How Has This Been Tested?
It's working on my server now, and a few other people I helped today. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

Adding large block for httpd nginx authentication with reverse proxy

Credit to TeraJL for the example .conf that got my reverse proxy working, as well!

### CODE REVIEW

**Comment by cmachikas** [comment]
Feedback would be great!

**Comment by FrostTheFox** [comment]
Looks pretty good to me at a glance.

**Comment by Ashex** [comment]
The use of ufw/firewall rules is not necessary. Instead include an item explaining that you should set `--host 127.0.0.1` so it only accepts local requests.

File: docs/extras/nginx.md [hunk 1]
```diff
@@ -71,3 +71,111 @@ server {
     }
 }
 ```
+
+##Adding simple httpd Authentication.
```
Comment (chuyskywalker) [inline docs/extras/nginx.md]: sphinx's markdown is kinda picky, you need a space after `##` here

File: docs/extras/nginx.md [hunk 2]
```diff
@@ -71,3 +71,111 @@ server {
...
+This will guide you through setting up simple HTTP Authentication using nginx and reverse proxy protocols. These instructions are written for someone using a Debian/Ubuntu VPS. Your enviroment may have slightly different requirements, however the concepts as a whole should still stand. This guide assumes you have nginx installed and running, and a `conf.d/*.conf` file created, such as `/etc/nginx/conf.d/pokemongo-map.conf`, as the example above provides, and that you're running your service on port 5000, and want it to be accessable at http://your_ip/go/, although it supports other ports and locations.  
+
+`*` denotes a wildcard, and will be used to stand for your site's `*.conf` file, please __do not__ literally type `sudo nano /etc/nginx/conf.d/*.conf`. 
+
+1. Create a .htpasswd file inside `/etc/nginx/`. Some suggested methods to create a .htpasswd file are below.
+   - Linux users can use the apache2-tools package to create the files.  
+ 
+      First, get the apache2-utils package
+
+      ``` sudo apt-get install apache2-utils   ```
```
Comment (chuyskywalker) [inline docs/extras/nginx.md]: Proper line breaks and indentation here and the next code block, please.

File: docs/extras/nginx.md [hunk 3]
```diff
@@ -71,3 +71,111 @@ server {
...
+      Troubleshooting:
+
+      * __I can't reach my server at http://your_ip/go/!__
+      
+      Check http://your_ip:5000/. If you cannot see it there, your issue lies with your server, not with nginx! If you can see it there, but cannot see it at http://your_ip/go/, your issue lies with nginx. Please check your config files to make sure they are pointing at the right places, and that your `sudo nginx -t` checks out. 
+      
+      * __nginx -t doesn't check out.__
+      
+      Check the error messages for which line is creating the error, and work your way backwards from there. Many times it's just a missed `;` or `}`. 
+
```
Comment (chuyskywalker) [inline docs/extras/nginx.md]: The troubleshooting section would work better as:

```
      Troubleshooting:

      - **I can't reach my server at http://your_ip/go/!**

         Check http://your_ip:5000/. If you cannot see it there, your issue lies with your server, not with nginx! If you can see it there, but cannot see it at http://your_ip/go/, your issue lies with nginx. Please check your config files to make sure they are pointing at the right places, and that your `sudo nginx -t` checks out.

      - **nginx -t doesn't check out.**

         Check the error messages for which line is creating the error, and work your way backwards from there. Many times it's just a missed `;` or `}`.
```

File: docs/extras/nginx.md [hunk 4]
```diff
@@ -71,3 +71,111 @@ server {
...
+5. Use ufw or another service to block direct access to port 5000. The authentication wouldn't be very good if you could bypass it by appending a :5000 to the url, so we're going to have to shut down port 5000 access remotely, only allowing internal services, IE nginx, to access it. We're going to assume, for the purposes of this tutorial, that you are going with ufw, but many firewall options exist! First step is to verify ufw is installed. 
+
+   ```sudo apt-get install ufw```
```
Comment (chuyskywalker) [inline docs/extras/nginx.md]: Again, proper linebreak needed here.

File: docs/extras/nginx.md [hunk 5]
```diff
@@ -71,3 +71,111 @@ server {
...
+   sudo ufw default allow outgoing
+   sudo ufw default deny incoming
+   sudo ufw deny 5000
+   ```
+   
+   If you want any additional information on how to configure more ufw services, please refer to the following guide: https://www.digitalocean.com/community/tutorials/how-to-setup-a-firewall-with-ufw-on-an-ubuntu-and-debian-cloud-server 
+   
+   Finally, we're going to enable ufw using the command 
+   
+   ``` sudo ufw enable ```
```
Comment (chuyskywalker) [inline docs/extras/nginx.md]: Ditto

**Comment by chuyskywalker** [comment]
^ That comment is a good idea (and is less platform specific which is nice) and needs some minor formatting clean up items addressed.

Thanks!

**Comment by cmachikas** [comment]
Sounds good, I'll find some time this afternoon to make the appropriate changes. Thanks for the non-firewall suggestion, as well!

**Comment by cmachikas** [comment]
Pushed a new commit 4ced11f

**Pull request merged**